CMakeLists.txt

```diff
@@ -1,11 +1,144 @@
 cmake_minimum_required(VERSION 3.9)
-project(readerwriterqueue VERSION 1.0.0)
+set(CMAKE_CXX_STANDARD 11)  # don't need to specify manually per target anymore.
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include(GNUInstallDirs)
 
-add_library(${PROJECT_NAME} INTERFACE)
+project(readerwriterqueue)
 
-target_include_directories(readerwriterqueue INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+add_library(readerwriterqueue INTERFACE)
 
-install(FILES atomicops.h readerwriterqueue.h readerwritercircularbuffer.h LICENSE.md
+# This needs to be updated everytime the library tag version updates. The version is not 1.0.0, it's 1.0.6 according to tags.
+set_target_properties(readerwriterqueue PROPERTIES
+        SOVERSION 1
+        VERSION 1.0.6)
+
+#see this talk for why these changes need to be made https://www.youtube.com/watch?v=m0DwB4OvDXk
```

Comment:
Should this be a comment on the PR instead of in the source code?

Reply:
No,  that explains the idea behind the commands used there, I wasn't talking about the pull request rational, and now I realize that's poorly worded, CMake documentation does not explain all the rational behind what installable functions to use where that I made here,  and that's unfortunately the best place to find the kind of information I used here.  i